File: project_euler_002.rb
def even_fib_calc num
	even_fib_array = []
	fib_array = [1,2,3]
	count = 2

	until(fib_array.last > num)
		fib_array.push(fib_array[count] + fib_array[count-1])
		even_fib_array << fib_array[count] if fib_array[count] % 2 == 0
		count += 1
	end

	return even_fib_array.inject(:+) + 2
end

puts even_fib_calc(4_000_000)

File: project_euler_002.py
def even_fib_calc( number ):
	even_fib_list = []
	fib_list = [1,2,3]
	count = 2

	while(fib_list[-1] < number):
		fib_list.append(fib_list[count] + fib_list[count-1])
		if (fib_list[count] % 2 == 0):
			even_fib_list.append(fib_list[count])
		count += 1
	
	return sum(even_fib_list + 2)

print(even_fib_calc(4000000))